File: Fundamentos_da_Computacao/10-04/Intersecao_Arrays.py
# Imprimir a interseção de dois vetores

vetor1 = [0]*3
vetor2 = [0]*3
vetor_intersecao = []

for i in range(len(vetor1)):
    vetor1[i] = int(input("Escreva um valor para Vetor1 na coordenada {}\n".format(i)))
    vetor2[i] = int(input("Escreva um valor para Vetor2 na coordenada {}\n".format(i)))
    
for x in vetor1:
    if (x in vetor2) and not (x in vetor_intersecao):
        vetor_intersecao.append(x)

print("Vetor1: {} \nVetor2: {} \nInterceção dos vetores: {}".format(vetor1, vetor2, vetor_intersecao))

File: Fundamentos_da_Computacao/10-04/Intersecao_Arrays.rs
//EM PRODUÇÃO
use std::io;

fn main() {
    let mut vetor1: [i32; 5] = [0; 5];
    let mut vetor2: [i32; 5] = [0; 5];
    let mut vetor_intersecao [i32; 5] = [0; 5];
    
    for i in 0..5 {
        vetor1[i] = ler_i32(format!("Digite um valor para o Vetor 01 na coordenada {}", i));
        vetor2[i] = ler_i32(format!("Digite um valor para o Vetor 02 na coordenada {}", i));
    }
    
    for x in vetor1 {
        for (x in vetor2) and !(x in vetor_intersecao) {
            d
        }
    }
    println!("{:?} \n{:?}", vetor1, vetor2);
}

fn ler_i32(texto: String) -> i32 {
    println!("{}", texto);
    
    let mut input = String::new();
        io::stdin().read_line(&mut input).expect("Erro ao ler a linha!");
    
    input.trim().parse().expect("Erro ao converter o input")
}
